jovo-integrations/jovo-db-firestore/src/Firestore.ts

```diff
@@ -20,7 +20,10 @@ export class Firestore implements Db {
     firebaseAdmin?: any; // tslint:disable-line
     firestore?: firebase.firestore.Firestore;
 
-    constructor(config?: Config) {
+    constructor(firestore?: firebase.firestore.Firestore, config?: Config) {
```

Comment:
Thank you for catching that. I have switched the order of the parameters in the constructor to ensure that existing applications don't break.

I like your second idea as well, however skipping internal initialization seemed simpler if the instance was passed in the constructor rather than through a setter.